web/src/components/sign-up/sign-up.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import StateTree from '../../stores/tree';
+import API from '../../services/api';
+import { connect } from 'react-redux';
+import { CheckIcon, OldPlayIcon, CautionIcon } from '../ui/icons';
+
+import './sign-up.css';
+
+interface PropsFromState {
+  api: API;
+}
+
+interface State {
+  email: string;
+  buttonStyle: string;
+  buttonHtml: any;
+}
+
+class SignUp extends React.Component<PropsFromState, State> {
+  constructor(props: PropsFromState, context: any) {
+    super(props, context);
+    const { api } = this.props;
+    this.state = {
+      email: '',
+      buttonStyle: 'submit-button',
+      buttonHtml: <OldPlayIcon className="icon" />,
+    };
+    this.handleChange = this.handleChange.bind(this);
```

Comment:
nit: in other components we tend to do binding by assigning functions to properties, e.g. `handleChange = (event) => {}` (similar for state, then you won't need a constructor).